Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -254,16 +321,40 @@ open class ChatMessageListVC: _ViewController,
         listView.scrollToMostRecentMessage(animated: animated)
     }
 
-    func updateUnreadMessagesSeparator(at id: MessageId?, previousId: MessageId?) {
+    func updateUnreadMessagesSeparator(at lastReadId: MessageId?) {
+        let previousLastReadId = lastReadMessageId
+        guard previousLastReadId != lastReadId else { return }
+
         func indexPath(for id: MessageId?) -> IndexPath? {
             id.flatMap(getIndexPath)
         }
 
-        let indexPathsToReload = [indexPath(for: previousId), indexPath(for: id)].compactMap { $0 }
+        lastReadMessageId = lastReadId
+        let indexPathsToReload = [indexPath(for: previousLastReadId), indexPath(for: lastReadId)].compactMap { $0 }
```

Comment:
Can we  reuse the `getIndexPath(forMessageId:)` method rather than having a new one?